CUDADataFormats/TrackingRecHit/test/TrackingRecHit2DCUDA_t.cpp

```diff
@@ -16,18 +16,25 @@ int main() {
   cudaCheck(cudaStreamCreateWithFlags(&stream, cudaStreamNonBlocking));
 
   auto nHits = 200;
-  auto nModules = 2000;
   // inner scope to deallocate memory before destroying the stream
   {
-    TrackingRecHit2DGPU tkhit(nHits, nModules, 0, nullptr, nullptr, stream);
-
+    TrackingRecHit2DGPU tkhit(nHits, false, 0, nullptr, nullptr, stream);
     testTrackingRecHit2D::runKernels(tkhit.view());
 
-    TrackingRecHit2DHost tkhitH(nHits, nModules, 0, nullptr, nullptr, stream, &tkhit);
+    TrackingRecHit2DGPU tkhitPhase2(nHits, true, 0, nullptr, nullptr, stream);
+    testTrackingRecHit2D::runKernels(tkhitPhase2.view());
+
+    TrackingRecHit2DHost tkhitH(nHits, false, 0, nullptr, nullptr, stream, &tkhit);
     cudaStreamSynchronize(stream);
     assert(tkhitH.view());
     assert(tkhitH.view()->nHits() == unsigned(nHits));
-    assert(tkhitH.view()->nMaxModules() == unsigned(nModules));
+    assert(tkhitH.view()->nMaxModules() == 1856);
```

Comment:
shouldn't this be `phase1PixelTopology::numberOfModules`?
Similar for the phase2 case below